src/OpenTelemetry/Program.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System.Diagnostics.Metrics;
+using System.Reflection;
+using Microsoft.Extensions.Logging;
+using OpenTelemetry;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+using Temporalio.Client;
+using Temporalio.Extensions.DiagnosticSource;
+using Temporalio.Extensions.OpenTelemetry;
+using Temporalio.Runtime;
+using Temporalio.Worker;
+using TemporalioSamples.OpenTelemetry;
+
+AssemblyName assemblyName = typeof(TemporalClient).Assembly.GetName();
+
+using var meter = new Meter(assemblyName.Name!, assemblyName.Version!.ToString());
+
+string instanceId = args.ElementAtOrDefault(0) ?? throw new ArgumentException("Must pass 'worker' or 'workflow' as the single argument");

```

Comment:
A little bit of repetition here, but it matters greatly for the OpenTelemetry sample in terms of correlating telemetry coming from the worker or client.